crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -147,6 +157,48 @@ impl<'a> LimitsTraversal<'a> {
         Ok(())
     }
 
+    fn check_multiget_queries(
+        &mut self,
+        op: &'a Positioned<OperationDefinition>,
+    ) -> ServerResult<()> {
+        for item in &op.node.selection_set.node.items {
+            self.traverse_multiget_queries(item)?;
+        }
+        Ok(())
+    }
+
+    fn traverse_multiget_queries(&mut self, item: &'a Positioned<Selection>) -> ServerResult<()> {
+        match &item.node {
+            Selection::Field(f) => {
+                let name = &f.node.name.node;
+                if name.starts_with(MULTI_GET_OBJECTS) {
+                    for (name, value) in &f.node.arguments {
+                        self.check_multiget_objs_arg(&name.node, value)?;
+                    }
+                }
+            }
+
+            Selection::InlineFragment(f) => {
+                for selection in &f.node.selection_set.node.items {
+                    self.traverse_multiget_queries(selection)?;
+                }
+            }
+
+            Selection::FragmentSpread(fs) => {
+                let name = &fs.node.fragment_name.node;
+                let def = self
+                    .fragments
+                    .get(name)
+                    .expect("Fragment not found in document");
```

Comment:
We cannot use `expect` here -- it will cause the entire service to panic if someone passes in a malformed query! Everywhere else, we use `.ok_or_else`.

Reply:
🤦 sorry, you're right.